pilot/pkg/xds/filters/filters.go

```diff
@@ -291,6 +292,80 @@ var (
 		},
 	}
 
+	// GatewayMtlsMetadataFilter is configured on gateways when the TLS mode for listener is ISTIO_MUTUAL
+	// This allows egress gateways to discover and propagate metadata exchange headers with the downstream peer.
+	GatewayMtlsMetadataFilter = &hcm.HttpFilter{
+		Name: MxFilterName,
+		ConfigType: &hcm.HttpFilter_TypedConfig{
+			TypedConfig: protoconv.TypedStructWithFields("type.googleapis.com/io.istio.http.peer_metadata.Config",
+				map[string]any{
+					"upstream_discovery": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+						map[string]any{
+							"workload_discovery": map[string]any{},
+						},
+					},
+					"upstream_propagation": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+					},
+					"downstream_discovery": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+						map[string]any{
+							"workload_discovery": map[string]any{},
+						},
+					},
+					"downstream_propagation": []any{
+						map[string]any{
+							"istio_headers": map[string]any{},
+						},
+					},
+				}),
```

Comment:
Sorry about a late response. I think it's not a good idea to generate stats that are not pinned to the gateway like this config implies. This creates a high risk of cardinality explosion since for any client M connected to the gateway that routes to a server N, you'd need M*N pairs of metadata. It'd be better to keep it consistent and create telemetry with gateway as either source of destination.